deploy-service/common/src/main/java/com/pinterest/deployservice/common/EncryptionUtils.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Date;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+/**
+ * Utility class for loading certificates and keys.
+ * based on: https://github.com/Mastercard/client-encryption-java/blob/main/src/main/java/com/mastercard/developer/utils/EncryptionUtils.java
+ */
+public final class EncryptionUtils {
+
+    private static final String PKCS_1_PEM_HEADER = "-----BEGIN RSA PRIVATE KEY-----";
+    private static final String PKCS_1_PEM_FOOTER = "-----END RSA PRIVATE KEY-----";
+    private static final String PKCS_8_PEM_HEADER = "-----BEGIN PRIVATE KEY-----";
+    private static final String PKCS_8_PEM_FOOTER = "-----END PRIVATE KEY-----";
+
+    private EncryptionUtils() {
+    }
+
+    public static String createGithubJWT(String githubAppId, String privateKeyPEMString, long ttlMillis) throws Exception {
+        // The JWT signature algorithm we will be using to sign the token
+        SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.RS256;
+
+        // We will sign our JWT with our private key
+        Key signingKey = EncryptionUtils.loadDecryptionKey(privateKeyPEMString);
+
+        long nowMillis = System.currentTimeMillis();
+        Date now = new Date(nowMillis);
+
+        // Let's set the JWT Claims
+        JwtBuilder builder = Jwts.builder().setIssuedAt(now).setIssuer(githubAppId).signWith(signingKey,
+            signatureAlgorithm);
+
+        // if it has been specified, let's add the expiration
+        if (ttlMillis > 0) {
+            long expMillis = nowMillis + ttlMillis;
+            Date exp = new Date(expMillis);
+            builder.setExpiration(exp);
+        }
+
+        // Builds the JWT and serializes it to a compact, URL-safe string
+        return builder.compact();
+    }
+
+    /**
+     * Load a RSA decryption key from a file (PEM or DER).
+     */
+    public static PrivateKey loadDecryptionKey(String keyString) throws GeneralSecurityException, IOException {
+        if (keyString.contains(PKCS_1_PEM_HEADER)) {
+            // OpenSSL / PKCS#1 Base64 PEM encoded file
+            keyString = keyString.replace(PKCS_1_PEM_HEADER, "");
+            keyString = keyString.replace(PKCS_1_PEM_FOOTER, "");
+            keyString = keyString.replace("\n", "");
+            keyString = keyString.replace("\r\n", "");
+            return readPkcs1PrivateKey(Base64.getDecoder().decode(keyString));
+        }
+        else if (keyString.contains(PKCS_8_PEM_HEADER)) {
+            // PKCS#8 Base64 PEM encoded file
+            keyString = keyString.replace(PKCS_8_PEM_HEADER, "");
+            keyString = keyString.replace(PKCS_8_PEM_FOOTER, "");
+            keyString = keyString.replace("\n", "");
+            keyString = keyString.replace("\r\n", "");
+            return readPkcs8PrivateKey(Base64.getDecoder().decode(keyString));
+        }
+        else {
+            throw new IllegalArgumentException("Unexpected key format!");
```

Comment:
Can we also log all the exceptions?

Reply:
added logging in caller githubmanager.java